packages/patrol/test/patrol_tester_test.dart

```diff
@@ -936,5 +937,104 @@ void main() {
         expect($('bottom text').visible, true);
       });
     });
+
+    group('pumpAndMaybeSettle()', () {
+      late int count;
+      late int state;
+
+      setUp(() {
+        count = 0;
+        state = 0;
+      });
+
+      final appWithInfiniteAnimation = MaterialApp(
+        home: Scaffold(
+          body: Center(
+            child: StatefulBuilder(
+              builder: (context, setState) {
+                return Column(
+                  children: [
+                    CircularProgressIndicator(),
+                    Text('count: $count'),
+                    ElevatedButton(
+                      onPressed: () => setState(() => count++),
+                      style: ElevatedButton.styleFrom(
+                        textStyle: const TextStyle(fontSize: 20),
+                      ),
+                      child: const Text('Enabled button'),
+                    ),
+                  ],
+                );
+              },
+            ),
+          ),
+        ),
+      );
+
+      final appWithAnimationOnTap = MaterialApp(
+        home: Scaffold(
+          body: Center(
+            child: StatefulBuilder(
+              builder: (context, setState) {
+                return Column(
+                  children: [
+                    Text('count: $count'),
+                    ElevatedButton(
+                      onPressed: () => setState(() {
+                        state = 1;
+                        Timer(Duration(seconds: 5), () {
+                          setState(() {
+                            state = 0;
+                            count++;
+                          });
+                        });
+                      }),
+                      style: ElevatedButton.styleFrom(
+                        textStyle: const TextStyle(fontSize: 20),
+                      ),
+                      child: state != 0
+                          ? CircularProgressIndicator()
+                          : const Text('Enabled button'),
+                    ),
+                  ],
+                );
+              },
+            ),
+          ),
+        ),
+      );
+
+      patrolTest(
+          "doesn't throw an exception when trying to settle an infinite amimation",
+          ($) async {
+        await $.pumpWidget(appWithInfiniteAnimation);
+
+        await $(ElevatedButton).tap(settleBehavior: SettleBehavior.maybeSettle);
+
+        expect($('count: 1'), findsOneWidget);
```

Comment:
We should also verify that the call to `tap()` took exactly 10 seconds.

I think the [fake_async](https://pub.dev/packages/fake_async) package can be used for that.

Reply:
Would `stopwatch` be enough? Just asking

Reply:
Does it support mocking time passage? If yes that's fine.

The 2 requirements for this is:
- verify that the call to tap() took exactly 10 seconds
- mock time passage so the test doesn't actually take 10 seconds